keras_cv/layers/preprocessing/random_hue.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv.utils import preprocessing
+
+
+class RandomHue(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Randomly adjusts the hue on given images.
+
+    This layer will randomly increase/reduce the hue for the input RGB
+    images. At inference time, the output will be identical to the input.
+    Call the layer with `training=True` to adjust the brightness of the input.
+
+    The image hue is adjusted by converting the image(s) to HSV and rotating the
+    hue channel (H) by delta. The image is then converted back to RGB.
+
+    Args:
+        factor: Either a tuple of two floats or a single float. `factor` controls the
+            extent to which the image saturation is impacted. `factor` =
+            `0.0`, `0.5` or `1.0` makes this layer perform a no-op operation.
```

Comment:
> 0.0`, `0.5` or `1.0` makes this layer perform a no-op operation.
is this true for all three values?  isn't that a little strange an unexpected?

Reply:
ah - I see this is because of the way we transform the value range.

Reply:
yea, the hue adjustment is like a cycle. When u adjust to the max or min, it comes back to the original state. I could change the mapping to [-0.5, 0.5] so that the the input [0, 1] could achieve the max opposite hue value.

Let me know what you prefer.

Reply:
Note that even we change the value mapping, the 0 and 1 will achieve the max adjustment, but the output will be the same between 0 and 1, (since the adjustment is a cycle, 0 and 1 just go from different direction.)